src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,52 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d.MM.yyyy");
+    private static final int DATE = 0;
+    private static final int NAME = 1;
+    private static final int WORK_HOURS = 2;
+    private static final int PAYMENT = 3;
+
+    public static String getSalaryInfo(String[] names, String[] data, String dateFrom,
+                                       String dateTo) {
+
+        LocalDate from = LocalDate.parse(dateFrom, formatter);
+        LocalDate to = LocalDate.parse(dateTo, formatter);
+        StringBuilder reportForPeriod = new StringBuilder()
+                .append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
```

Comment:
let's remove adding lineSeparator from here, and will add it before each name
you won't need to delete anything from stringBuilder in the end